app/helpers/course_summaries_helper.rb

```diff
@@ -0,0 +1,156 @@
+
+      {
+        id: student.id,
+        user_name: student.user_name,
+        first_name: student.first_name,
+        last_name: student.last_name,
+        assignment_marks: marks,
+        grade_entry_form_marks: gef_marks,
+        weighted_marks:
+          get_weighted_total_for_all_marking_schemes_for_student(
+            all_schemes, marks, gef_marks, weights, gef_weights, max_marks,
+            gef_max_marks)
+      }
+    end
+    student_list.to_json
+  end
+
+  # Get marks for all assignments for a student
+  def get_mark_for_all_assignments_for_student(student, assignments)
+    marks = Hash[assignments.map {|a| [a.id, 0]}]
+
+    student.groupings.each do |g|
+      sub = g.current_submission_used
+      marks[g.assignment_id] = sub ? sub.get_latest_result.total_mark : 0
+    end
+
+    marks
+  end
+
+  # Get max mark for assignment with id a_id
+  def get_max_mark_for_assignment(a_id)
+    max_mark = 0
+    RubricCriterion.where(assignment_id: a_id).each do |rc|
+      max_mark += rc.weight * 4
+    end
+
+    FlexibleCriterion.where(assignment_id: a_id).each do |fc|
+      max_mark += fc.max
+    end
+    max_mark
+  end
+
+  def get_mark_for_all_gef_for_student(student, gefs)
+    gef_marks = Hash[gefs.map {|gef| [gef.id, 0]}]
+    student.grade_entry_students.each do |ges|
+      gef_marks[ges.grade_entry_form_id] = ges.total_grade.nil? ? 0 : ges.total_grade
+    end
+    gef_marks
+  end
+
+  def get_max_mark_for_grade_entry_form(gef_id)
+    total = 0
+    GradeEntryItem.where(grade_entry_form_id: gef_id).each do |gei|
+      total += gei.out_of
+    end
+    total
+  end
+
+  # Return an object that contains a key for each marking
+  # scheme. And the value is an object with keys as assignment ids
+  # and the value the weight for that assignment
+  def get_marking_weights_for_all_marking_schemes
+    result = {}
+    MarkingScheme.all.each do |ms|
+      result[ms.id] = {}
+      MarkingWeight.where(marking_scheme_id: ms.id, is_assignment: true).each do |mw|
+        result[ms.id][mw.gradable_item_id] = mw.weight
+      end
+    end
+    result
+  end
+
+  def get_gef_marking_weights_for_all_marking_schemes
+    result = {}
+    MarkingScheme.all.each do |ms|
+      result[ms.id] = {}
+      MarkingWeight.where(marking_scheme_id: ms.id, is_assignment: false).each do |mw|
+        result[ms.id][mw.gradable_item_id] = mw.weight
+      end
+    end
+    result
+  end
+
+  def get_weighted_total_for_all_marking_schemes_for_student(
+    schemes,
+    assignment_marks, gef_marks,
+    weights, gef_weights,
+    max_marks, gef_max_marks)
+
+    result = {}
+    schemes.each do |ms|
+      result[ms.id] =
+       get_weighted_total_for_marking_scheme_and_student(
+         weights[ms.id], gef_weights[ms.id],
+         assignment_marks, gef_marks,
+         max_marks, gef_max_marks)
+    end
+    result
+  end
+
+  def get_weighted_total_for_marking_scheme_and_student(
```

Comment:
Assignment Branch Condition size for get_weighted_total_for_marking_scheme_and_student is too high. [17.32/15]